internal/provider/resource_rpaas_route.go

```diff
@@ -66,40 +67,61 @@ func resourceRpaasRoute() *schema.Resource {
 	}
 }
 
-func resourceRpaasRouteUpsert(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func resourceRpaasRouteCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	provider := meta.(*rpaasProvider)
 
+	instance := d.Get("instance").(string)
 	serviceName := d.Get("service_name").(string)
+	path := d.Get("path").(string)
+
 	rpaasClient, err := provider.RpaasClient.SetService(serviceName)
 	if err != nil {
 		return diag.Errorf("Unable to create client for service %s: %v", serviceName, err)
 	}
 
-	instance := d.Get("instance").(string)
-	path := d.Get("path").(string)
-	httpsOnly := d.Get("https_only").(bool)
-	args := rpaas_client.UpdateRouteArgs{
-		Instance:  instance,
-		Path:      path,
-		HTTPSOnly: httpsOnly,
+	tflog.Info(ctx, "Create route", map[string]interface{}{
+		"service":  serviceName,
+		"instance": instance,
+		"path":     path,
+	})
+
+	err = rpaasRetry(ctx, d, func() error {
+		return updateRpaasRoute(ctx, d, instance, path, rpaasClient)
+	})
+	if err != nil {
+		return diag.Errorf("Unable to create route %s for instance %s: %v", path, instance, err)
 	}
 
-	if content, ok := d.GetOk("content"); ok {
-		args.Content = content.(string)
+	d.SetId(fmt.Sprintf("%s::%s::%s", serviceName, instance, path))
+	return resourceRpaasRouteRead(ctx, d, meta)
+}
+
+func resourceRpaasRouteUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	provider := meta.(*rpaasProvider)
+
+	serviceName, instance, path, err := parseRpaasRouteID(d.Id())
+	if err != nil {
+		return diag.Errorf("Unable to parse Route ID: %v", err)
 	}
 
-	if destination, ok := d.GetOk("destination"); ok {
-		args.Destination = destination.(string)
+	rpaasClient, err := provider.RpaasClient.SetService(serviceName)
+	if err != nil {
+		return diag.Errorf("Unable to create client for service %s: %v", serviceName, err)
 	}
 
+	tflog.Info(ctx, "Update route", map[string]interface{}{
+		"service":  serviceName,
+		"instance": instance,
+		"path":     path,
+	})
+
 	err = rpaasRetry(ctx, d, func() error {
-		return rpaasClient.UpdateRoute(ctx, args) // UpdateRoute is really an Upsert
+		return updateRpaasRoute(ctx, d, instance, path, rpaasClient)
 	})
 	if err != nil {
-		return diag.Errorf("Unable to create/update route %s for instance %s: %v", path, instance, err)
+		return diag.Errorf("Unable to create route %s for instance %s: %v", path, instance, err)
```

Comment:
`"Unable to create route` should be `"Unable to update route`

Reply:
thanks